src/createWagmiConfig.ts

```diff
@@ -0,0 +1,37 @@
+import { http, cookieStorage, createConfig, createStorage } from 'wagmi';
+import { base, baseSepolia } from 'wagmi/chains';
+import { coinbaseWallet } from 'wagmi/connectors';
+import type { CreateWagmiConfigParams } from './types';
+
+// createWagmiConfig returns a WagmiConfig (https://wagmi.sh/react/api/createConfig) using OnchainKit provided settings.
+// This function should only be used if the user does not provide WagmiProvider as a parent in the React context.
+export const createWagmiConfig = ({
+  apiKey,
+  appName,
+  appLogoUrl,
+}: CreateWagmiConfigParams) => {
+  return createConfig({
+    chains: [base, baseSepolia],
+    connectors: [
+      coinbaseWallet({
+        appName,
+        appLogoUrl,
+        preference: 'smartWalletOnly',
```

Comment:
should we set this to `all`?

Reply:
seems like it'd be a more flexible implementation, unless we just really want to push to smart wallet.

Reply:
we need to add a second connector for EOA 
`
    coinbaseWallet({
      appName: 'OnchainKit',
      preference: 'eoaOnly',
    }),
`